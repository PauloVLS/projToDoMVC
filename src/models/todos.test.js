import { getSnapshot, onSnapshot, onPatch } from "mobx-state-tree";
import { Todo, TodoStore } from "./Todos";

it("can create  an instace of a model", () => {
    const item = Todo.create({
        text: "teste",
        completed: false,
        id: 1
    })

    expect(item.text).toBe("teste")
    expect(item.completed).toBe(false)
    item.edit("teste2")
    expect(item.text).toBe("teste2")
    item.toggle()
    expect(item.completed).toBe(true)
})

it("can create a todo list", () => {
    const list = TodoStore.create({
        todos: [
            {
                text: "teste",
                completed: false,
                id: 1
            }
        ]
    })

    expect(list.todos.length).toBe(1)
    expect(list.todos[0].completed).toBe(false)
})

it("can add new items - 2", () => {
    const list = TodoStore.create()
    const patches = []
    onPatch(list, patch => {
        patches.push(patch)
    })

    list.addTodo("teste2")

    list.todos[0].edit("teste3")

    expect(patches).toMatchSnapshot()
})

it("can add new items", () => {
    const list = TodoStore.create()
    const states = []
    onSnapshot(list, snapshot => {
        states.push(snapshot)
    })

    list.addTodo("teste2")

    expect(list.todos.length).toBe(1)
    expect(list.todos[0].text).toBe("teste2")
    list.todos[0].edit("teste3")
    expect(list.todos[0].text).toBe("teste3")

    expect(getSnapshot(list)).toMatchSnapshot()

    expect(states).toMatchSnapshot()
})

it("can remove items", () => {
    const list = TodoStore.create({
        todos: [
            {
                text: "learn Mobx",
                completed: false,
                id: 0
            },
            {
                text: "learn MST",
                completed: true,
                id: 1
            }
        ]
    })

    expect(list.filteredTodos[0].text).toBe("learn Mobx")
    list.todos[0].remove()
    expect(list.filteredTodos[0].text).toBe("learn MST")
})

it("can view and filter Todos", () => {
    const list = TodoStore.create({
        todos: [
            {
                text: "learn Mobx",
                completed: false,
                id: 0
            },
            {
                text: "learn MST",
                completed: true,
                id: 1
            }
        ]
    })

    expect(list.activeCount).toBe(1)
    expect(list.completedCount).toBe(1)
    expect(list.filteredTodos[0].text).toBe("learn Mobx")
    list.setFilter("show_completed")
    expect(list.filteredTodos[0].text).toBe("learn MST")
})