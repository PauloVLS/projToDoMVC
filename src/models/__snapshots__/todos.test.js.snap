// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add new items - 2 1`] = `
Array [
  Object {
    "op": "add",
    "path": "/todos/0",
    "value": Object {
      "completed": false,
      "id": 0,
      "text": "teste2",
    },
  },
  Object {
    "op": "replace",
    "path": "/todos/0/text",
    "value": "teste3",
  },
]
`;

exports[`can add new items 1`] = `
Object {
  "filter": "show_all",
  "todos": Array [
    Object {
      "completed": false,
      "id": 0,
      "text": "teste3",
    },
  ],
}
`;

exports[`can add new items 2`] = `
Array [
  Object {
    "filter": "show_all",
    "todos": Array [
      Object {
        "completed": false,
        "id": 0,
        "text": "teste2",
      },
    ],
  },
  Object {
    "filter": "show_all",
    "todos": Array [
      Object {
        "completed": false,
        "id": 0,
        "text": "teste3",
      },
    ],
  },
]
`;
